src/sindri/sindri.py

```diff
@@ -48,38 +46,103 @@ class APIError(Exception):
     DEFAULT_SINDRI_API_URL = "https://sindri.app/api/"
     API_VERSION = "v1"
 
-    def __init__(
-        self,
-        api_key: str,
-        api_url: str = DEFAULT_SINDRI_API_URL,
-        verbose_level: int = 0,
-    ):
-        """
-        Parameters
+    VERBOSE_LEVELS: list[int] = [0, 1, 2]
+
+    def __init__(self, api_key: str, verbose_level: int = 0, **kwargs):
+        """Initialize an instance of the Sindri SDK.
+
+        Args:
+        - `api_key`: Your Sindri API Key.
+        - `verbose_level`: Must be either `0`, `1`, or `2`.
+            - `0`: Do not print anything to stdout.
+            - `1`: Print only necesessary information from Circuit/Proof objects.
+            - `2`: Print everything.
 
-        - `api_key: str`: Sindri API Key
-        - `api_url: str`: Sindri API Url
-        - `verbose_level: int`: Stdout print level. Options=`[0,1,2]`
+        Returns:
+        - `sindri`: An instance of the class configured with your Sindri API Key.
+
+        Raises:
+        - `Sindri.APIError`:
+            - Your API Key is improperly formatted.
         """
         # Obtain version from module
         self.version = __version__
 
+        self.headers_json: dict = {}  # set in set_api_key()
+
         # Do not print anything during initial setup
         self.set_verbose_level(0)
 
         self.polling_interval_sec: int = 1  # polling interval for circuit compilation & proving
         self.max_polling_iterations: int = 172800  # 2 days with polling interval 1 second
         self.perform_verify: bool = False
-        self.set_api_url(api_url)
+
+        # Set API Url
+        api_url = kwargs.get("api_url", self.DEFAULT_SINDRI_API_URL)
+        if not isinstance(api_url, str):
+            raise Sindri.APIError("Invalid API Url")
+        if api_url == "":
+            raise Sindri.APIError("Invalid API Url")
+        if not api_url.endswith(self.API_VERSION) or not api_url.endswith(f"{self.API_VERSION}/"):
+            # Append f"{self.API_VERSION}/" to api_url
+            self._api_url = os.path.join(api_url, f"{self.API_VERSION}/")
```

Comment:
`self._api_url` is only set if the conditional fails, yet is referenced all over the following code. Should `self._api_url` be set after the conditional is handled?

Reply:
Apparently this commit is supplanted by 63c9766.